src/Symfony/Component/Security/Http/Authentication/AuthenticatorManager.php

```diff
@@ -69,7 +70,7 @@ public function __construct(iterable $authenticators, TokenStorageInterface $tok
     public function authenticateUser(UserInterface $user, AuthenticatorInterface $authenticator, Request $request, array $badges = []): ?Response
     {
         // create an authenticated token for the User
-        $token = $authenticator->createAuthenticatedToken($passport = new SelfValidatingPassport($user, $badges), $this->firewallName);
+        $token = $authenticator->createAuthenticatedToken($passport = new SelfValidatingPassport(new UserBadge(null, function () use ($user) { return $user; }), $badges), $this->firewallName);
```

Comment:
`new UserBadge($user->getUsername(),`?

Again, we need to be really clear about the *purpose* of this first argument to `new UserBadge()`. That will make these decisions more clear.